static/js/topic_generator.js

```diff
@@ -112,17 +117,17 @@ exports.filter = function (gen, filter_func) {
 
 exports.next_topic = function (streams, get_topics, has_unread_messages, curr_stream, curr_topic) {
     var stream_gen = exports.wrap(streams, curr_stream);
-
     function get_topic_gen(which_stream) {
         var gen;
 
         if (which_stream === curr_stream) {
-            gen = exports.wrap_exclude(get_topics(which_stream), curr_topic);
+            gen = exports.wrap_exclude(get_topics(which_stream), { stream : curr_stream, topic : curr_topic});
```

Comment:
stylistically, we generally write this as `{stream: curr_stream, topic: curr_topic}`